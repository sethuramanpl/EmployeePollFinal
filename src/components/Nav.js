import { Link } from "react-router-dom";
import {connect} from "react-redux";
import { handleLogout } from "../actions/authedUser";
import { useNavigate } from 'react-router-dom';

import React from 'react'

const Nav = ({dispatch, authedUser}) => {

    let navigate = useNavigate();

    const clickLogout = (e) => {
        e.preventDefault();
        dispatch(handleLogout());
        navigate("/Login");

    }

  return (
    <div className="nav">
        <ul>
            <li>
                <Link to="/home">Home</Link>
            </li>
            <li>
                <Link to="/leaderboard">Leaderboard</Link>
            </li>
            <li>
                <Link to="/new">New Poll</Link>
            </li>
            <span>User: {authedUser}</span>
            <li>
                <Link onClick={clickLogout}>Logout</Link>
            </li>
        </ul>
    </div>
  )
}

const mapStateToProps = ({authedUser}) => ({
    authedUser,
});
export default connect(mapStateToProps)(Nav)