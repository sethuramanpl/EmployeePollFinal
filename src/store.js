import authedUser from "./reducers/authedUser";
import users from "./reducers/users";
import questions from "./reducers/questions";
import { combineReducers } from "redux";
import { createStore } from "redux";

import reducer from "./reducers";
import middleware from "./middleware";

 combineReducers({
    users,
    questions,
    authedUser,
})

const store = createStore(reducer, middleware);
export default store;