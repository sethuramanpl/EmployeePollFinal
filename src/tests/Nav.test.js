import { render, screen , fireEvent} from '@testing-library/react';
import Nav from '../components/Nav';
import React from 'react';
import {Provider} from "react-redux";
import { MemoryRouter } from 'react-router-dom';
import {BrowserRouter} from "react-router-dom";
import store from '../store';

describe("Nav", () => {
  it('should display all the navigation links in the page', async () => {
      const {wrapper} = render(
        <MemoryRouter>
          <Provider store={store}>
                  <Nav />
          </Provider>
        </MemoryRouter>
      );
      expect(screen.getByText('Home')).toBeInTheDocument();
      expect(screen.getByText('Leaderboard')).toBeInTheDocument();
      expect(screen.getByText('New Poll')).toBeInTheDocument();
      expect(screen.getByText('Logout')).toBeInTheDocument();
  });
});

describe("NavLinks", ()=>{
    it("should verify snapshot", async () => {
        const {wrapper} = render(
            <MemoryRouter>
              <Provider store={store}>
                      <Nav />
              </Provider>
            </MemoryRouter>
          );
    expect(wrapper).toMatchSnapshot();
    })
})