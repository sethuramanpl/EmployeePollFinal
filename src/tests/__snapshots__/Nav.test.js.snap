// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks should verify snapshot 1`] = `undefined`;
